app.js

```diff
@@ -112,6 +112,36 @@ function roundCounter(){
   roundCounter.innerText = currentRound;
 }
 
+// function setItems()
+function setItems(key, value) {
+  let stringified = JSON.stringify(value);
+  localStorage.setItem(key, stringified);
+}
+
+// function getItems()
+function getItems(key) {
+  let stringified =localStorage.getItem(key);
+  let parsed = JSON.parse(stringified);
+  return parsed;
+}
+
+//get updated product array and round from local storage if they exist, reset survey after 25 rounds
+function onLoad(){
+  if (getItems('allProducts')) {
+    allProducts = getItems('allProducts');
+  }
+
+  if (getItems('round')) {
+    currentRound = getItems('round');
+  }
+
+  if (currentRound >= 25){
+    localStorage.removeItem('allProducts');
+    localStorage.removeItem('round');
+    imagesSection.removeEventListener('click', showNewImage);
```

Comment:
With the way your code is set up, a user will complete 25 rounds, and if they revisit the page, the first time this will happen, preventing most functionality of your page and deleting the local storage without reassigning it.